exonum/src/node/consensus.rs

```diff
@@ -238,12 +238,50 @@ impl NodeHandler {
         Ok(())
     }
 
+    /// Checks if propose is correct (doesn't contain invalid transactions), and then
+    /// broadcasts a prevote for this propose.
+    ///
+    /// Returns `true` if majority of prevotes is achieved, and returns `false` otherwise.
+    fn check_propose_and_broadcast_prevote(&mut self, round: Round, propose_hash: Hash) -> bool {
+        // Do not send a prevote if propose contains incorrect transactions.
+        let propose_state = self.state.propose(&propose_hash).unwrap();
+        if propose_state.has_invalid_txs() {
+            warn!("Denying sending a prevote for a propose with incorrect transactions");
```

Comment:
I wonder if we need to save this incorrect proposal for the futher analysis?

Reply:
Not sure about this. Can you explain use cases?